unit_tests/UnitTestCreateOnDevice.C

```diff
@@ -0,0 +1,91 @@
+#include <gtest/gtest.h>
+
+#include "UnitTestUtils.h"
+
+
+#include <string>
+#include <iostream>
+#include <vector>
+
+#include <KokkosInterface.h>
+#include "utils/CreateDeviceExpression.h"
+
+class deviceable {
+protected :
+  deviceable *DeviceCopy;
```

Comment:
Apologies if this is picking nits, but shouldn't class names start with uppercase letters, and member variables start with lowercase letters, and end in an underscore?

Overall, though, I quite agree with @alanw0 — this looks great.  Thanks for doing this, @overfelt!